File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        mx = 0
        m = {}
        nums.sort()
        for u in nums:
            x = u-1
            if x in m:
                m[u] = m[x] + 1
                
            else:
                m[u] =1
            mx = max(mx, m[u])
        return mx

File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int mx = 0;
        for(int i= 0; i < nums.size(); i++) {
            if(!i) {
                mx = nums[i];
            }
            else {
                mx--;
                if(mx < 0) return false;
                mx = max(mx, nums[i]);
            }
        }
        return true;
    }
};
